### ISSUE TITLE
Code is blurry on Mobile just with normal typing (Android)

### ISSUE THREAD

**Issue by sfdexter** [SpaceDice9/OdeScriptEditor#13]
Even using a the original ODE Script Editor Experience

https://devforum.roblox.com/t/ode-script-editor-a-simple-embedded-script-editor-entirely-in-roblox/2565346#:~:text=Uncopylocked%20Roblox%20Place,123

typing "local test = 123" begins to blur on my Pixel 7 (Android).
It clears up when the TextBox loses focus.

I guess it needs to use a different font?
How does it look on IPhone?

**Comment by sfdexter** [comment]
iPhone looks even worse in landscape. 
The entire UI slides upwards to accommodate for the keyboard, so a lot of the editor moves off the screen.
And the Rich overlay completely slides off of the TextBox...it's unusable.
iPhone portrait mode is better...not much blur, but sometimes entire UI also slides upwards. Maybe there's an iPhone setting for that to fix

For Android, a workaround is setting the TextBox TextTransparency = 1.
No more blurriness, but the tradeoff is no text cursor. It's usable, but not great.
I tried a different font on Android and that didn't make a difference.

Other solutions I'm considering:
1) Replace every character in the TextBox with a space character. The cursor will be visible and and no blurriness.
Not perfect because the text cursor will often be a bit mis-positioned because the underlying text in the TextBox is not aligned with the Rich Overlay view.

2) Using a Frame to look like a text cursor and positioning it based on clicks. Sounds painful, error prone, and then there would be the nightmare of selecting text and trying to change the cursor frame Size to overlay on the selected text area.

**Comment by SpaceDice9** [comment]
Yeah, Ode was not really designed with mobile in mind. Not sure how to fix something like this but it could be that TextBoxes are just implemented a little differently on mobile which fucks up the way Ode accommodates for it.

**Comment by sfdexter** [comment]
I found something related...someone found some workarounds, but its not just an ODE problem...Roblox TextBoxes have issues.
 
https://devforum.roblox.com/t/how-exactly-does-the-mobile-ui-shift-when-editing-textboxes/2622784

Also, on iPhone -- I haven't found a way to do a newline character. I might actually have to add a button to add newlines!
I wouldn't care so much except for 58% of traffic for 'Ultimate Arduino Builder' https://www.roblox.com/share?code=11f2a5a0194da64aaf33fc122774d3b0&type=ExperienceDetails  is from mobile/tablet 
Terrible to lose more than 1/2 because of these issues.

ODE is really cool and I hope to contribute significant updates so it can be used beyond its original intentions. It's a strong base on which to build more.
